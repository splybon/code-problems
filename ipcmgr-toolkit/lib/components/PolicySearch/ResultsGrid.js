"use strict";

exports.__esModule = true;

var _uiToolkit = require("@icg360/ui-toolkit");

var _ipcmgrToolkit = require("@package/ipcmgr-toolkit");

var _classnames = require("classnames");

var _classnames2 = _interopRequireDefault(_classnames);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _constants = require("./constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ResultsGrid = function (_Component) {
  _inherits(ResultsGrid, _Component);

  function ResultsGrid() {
    _classCallCheck(this, ResultsGrid);

    return _possibleConstructorReturn(this, _Component.apply(this, arguments));
  }

  ResultsGrid.prototype.render = function render() {
    var _props = this.props,
        isLoading = _props.isLoading,
        policies = _props.policies,
        onPolicySelect = _props.onPolicySelect,
        onSortSelect = _props.onSortSelect,
        sortField = _props.sortField,
        sortOrder = _props.sortOrder;

    return _react2.default.createElement(
      _uiToolkit.Table,
      {
        className: "search__results",
        hover: !!policies.length,
        "data-content-loading": isLoading
      },
      _react2.default.createElement(
        "thead",
        null,
        _react2.default.createElement(
          "tr",
          null,
          _constants.TABLE_HEADINGS.map(function (_ref) {
            var label = _ref.label,
                sortable = _ref.sortable,
                value = _ref.value;
            return _react2.default.createElement(
              "th",
              { key: label },
              !!policies.length && sortable ? _react2.default.createElement(
                "button",
                {
                  type: "button",
                  className: (0, _classnames2.default)("search__sort", "sortable", sortOrder === _constants.SORT_ASC && sortField === value && "sortable--asc", sortOrder === _constants.SORT_DESC && sortField === value && "sortable--desc"),
                  onClick: function onClick() {
                    return onSortSelect(value);
                  }
                },
                label
              ) : label
            );
          })
        )
      ),
      _react2.default.createElement(
        "tbody",
        null,
        policies.length ? policies.map(function (_ref2) {
          var id = _ref2.id,
              addresses = _ref2.addresses,
              carrierId = _ref2.carrierId,
              effectiveDate = _ref2.effectiveDate,
              _ref2$identifiers = _ref2.identifiers,
              policyNumber = _ref2$identifiers.policyNumber,
              quoteNumber = _ref2$identifiers.quoteNumber,
              insuredName = _ref2.insuredName,
              policyState = _ref2.policyState,
              productLabel = _ref2.productLabel;

          var street1 = addresses && addresses.property && addresses.property.street1;
          return _react2.default.createElement(
            "tr",
            {
              key: id,
              onClick: function onClick() {
                return onPolicySelect({
                  id: id,
                  identifiers: { policyNumber: policyNumber, quoteNumber: quoteNumber },
                  insuredName: insuredName
                });
              }
            },
            _react2.default.createElement(
              "td",
              null,
              quoteNumber
            ),
            _react2.default.createElement(
              "td",
              null,
              policyNumber
            ),
            _react2.default.createElement(
              "td",
              null,
              carrierId
            ),
            _react2.default.createElement(
              "td",
              null,
              productLabel
            ),
            _react2.default.createElement(
              "td",
              null,
              insuredName
            ),
            _react2.default.createElement(
              "td",
              null,
              street1
            ),
            _react2.default.createElement(
              "td",
              null,
              policyState
            ),
            _react2.default.createElement(
              "td",
              null,
              (0, _ipcmgrToolkit.printDate)(effectiveDate)
            )
          );
        }) : _react2.default.createElement(
          "tr",
          null,
          _react2.default.createElement(
            "td",
            { colSpan: "8" },
            _react2.default.createElement(
              "div",
              { className: "search__results__zero-state" },
              _react2.default.createElement(_uiToolkit.Icon, {
                name: "search",
                size: "large",
                style: { opacity: ".325" }
              }),
              _react2.default.createElement(
                "h3",
                null,
                "No results found"
              ),
              _react2.default.createElement(
                "p",
                null,
                "Use the search field above to find quotes and policies by ID, address, policyholder name, and more."
              )
            )
          )
        )
      )
    );
  };

  return ResultsGrid;
}(_react.Component);

ResultsGrid.propTypes = process.env.NODE_ENV !== "production" ? {
  isLoading: _propTypes2.default.bool.isRequired,
  policies: _propTypes2.default.array.isRequired,
  onPolicySelect: _propTypes2.default.func.isRequired,
  onSortSelect: _propTypes2.default.func.isRequired,
  sortField: _propTypes2.default.string.isRequired,
  sortOrder: _propTypes2.default.string.isRequired
} : {};
exports.default = ResultsGrid;
module.exports = exports["default"];