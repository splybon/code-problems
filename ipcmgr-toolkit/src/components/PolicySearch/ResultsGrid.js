import { Icon, Table } from "@icg360/ui-toolkit";
import { printDate } from "@package/ipcmgr-toolkit";
import classnames from "classnames";
import PT from "prop-types";
import React, { Component } from "react";
import { SORT_ASC, SORT_DESC, TABLE_HEADINGS } from "./constants";

class ResultsGrid extends Component {
  static propTypes = {
    isLoading: PT.bool.isRequired,
    policies: PT.array.isRequired,
    onPolicySelect: PT.func.isRequired,
    onSortSelect: PT.func.isRequired,
    sortField: PT.string.isRequired,
    sortOrder: PT.string.isRequired,
    requireQuery: PT.bool.isRequired
  };

  render() {
    const {
      isLoading,
      policies,
      onPolicySelect,
      onSortSelect,
      sortField,
      sortOrder,
      requireQuery
    } = this.props;
    return (
      <Table
        className="search__results"
        hover={!!policies.length}
        data-content-loading={isLoading}
      >
        <thead>
          {
            <tr>
              {TABLE_HEADINGS.map(({ label, sortable, value }) => (
                <th key={label}>
                  {!!policies.length && sortable ? (
                    <button
                      type="button"
                      className={classnames(
                        "search__sort",
                        "sortable",
                        sortOrder === SORT_ASC &&
                          sortField === value &&
                          "sortable--asc",
                        sortOrder === SORT_DESC &&
                          sortField === value &&
                          "sortable--desc"
                      )}
                      onClick={() => onSortSelect(value)}
                    >
                      {label}
                    </button>
                  ) : (
                    label
                  )}
                </th>
              ))}
            </tr>
          }
        </thead>
        <tbody>
          {policies.length ? (
            policies.map(
              ({
                id,
                addresses,
                carrierId,
                effectiveDate,
                identifiers: { policyNumber, quoteNumber },
                insuredName,
                policyState,
                productLabel
              }) => {
                const street1 =
                  addresses && addresses.property && addresses.property.street1;
                return (
                  <tr
                    key={id}
                    onClick={() =>
                      onPolicySelect({
                        id,
                        identifiers: { policyNumber, quoteNumber },
                        insuredName
                      })
                    }
                  >
                    <td>{quoteNumber}</td>
                    <td>{policyNumber}</td>
                    <td>{carrierId}</td>
                    <td>{productLabel}</td>
                    <td>{insuredName}</td>
                    <td>{street1}</td>
                    <td>{policyState}</td>
                    <td>{printDate(effectiveDate)}</td>
                  </tr>
                );
              }
            )
          ) : (
            <tr>
              <td colSpan="8">
                <div className="search__results__zero-state">
                  <Icon
                    name="search"
                    size="large"
                    style={{ opacity: ".325" }}
                  />
                  <h3>No results found</h3>
                  {requireQuery && (
                    <p>
                      Use the search field above to find quotes and policies by
                      ID, address, policyholder name, and more.
                    </p>
                  )}
                </div>
              </td>
            </tr>
          )}
        </tbody>
      </Table>
    );
  }
}

export default ResultsGrid;
