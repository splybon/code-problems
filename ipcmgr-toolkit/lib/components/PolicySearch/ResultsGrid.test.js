import { shallow } from "enzyme";
import React from "react";
import ResultsGrid from "./ResultsGrid";

const mockProps = {
  policies: [
    {
      id: "exampleid",
      identifiers: { quoteNumber: "quotenum", policyNumber: "policynum" },
      carrerId: "carrier id",
      productLabel: "product label",
      insuredName: "insuredName",
      addresses: { property: { street1: "street1" } },
      policyState: "policyState",
      effectiveDate: "12/02/03"
    }
  ],
  isLoading: false,
  sortOrder: "DESC",
  sortField: "effectiveDate",
  onPolicySelect: () => {},
  onSortSelect: () => {}
};

it("renders with results", () => {
  const wrapper = shallow(<ResultsGrid {...mockProps} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders with no results", () => {
  const wrapper = shallow(<ResultsGrid {...mockProps} policies={[]} />);
  expect(wrapper.exists(".search__results__zero-state")).toBeTruthy();
});
