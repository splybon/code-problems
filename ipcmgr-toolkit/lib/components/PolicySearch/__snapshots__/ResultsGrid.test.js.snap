// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with results 1`] = `
<Table
  bordered={false}
  bsClass="table"
  className="search__results"
  condensed={false}
  data-content-loading={false}
  hover={true}
  responsive={false}
  striped={false}
>
  <thead>
    <tr>
      <th
        key="Quote ID"
      >
        <button
          className="search__sort sortable"
          onClick={[Function]}
          type="button"
        >
          Quote ID
        </button>
      </th>
      <th
        key="Policy #"
      >
        <button
          className="search__sort sortable"
          onClick={[Function]}
          type="button"
        >
          Policy #
        </button>
      </th>
      <th
        key="Carrier ID"
      >
        Carrier ID
      </th>
      <th
        key="Product"
      >
        Product
      </th>
      <th
        key="Name"
      >
        <button
          className="search__sort sortable"
          onClick={[Function]}
          type="button"
        >
          Name
        </button>
      </th>
      <th
        key="Address"
      >
        Address
      </th>
      <th
        key="Status"
      >
        <button
          className="search__sort sortable"
          onClick={[Function]}
          type="button"
        >
          Status
        </button>
      </th>
      <th
        key="Effective"
      >
        <button
          className="search__sort sortable sortable--desc"
          onClick={[Function]}
          type="button"
        >
          Effective
        </button>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="exampleid"
      onClick={[Function]}
    >
      <td>
        quotenum
      </td>
      <td>
        policynum
      </td>
      <td />
      <td>
        product label
      </td>
      <td>
        insuredName
      </td>
      <td>
        street1
      </td>
      <td>
        policyState
      </td>
      <td>
        12/02/2003
      </td>
    </tr>
  </tbody>
</Table>
`;
